### DISCUSSION TITLE
Compressed list data structure

### DISCUSSION THREAD

**Discussion by jkanche** [BiocPy/BiocPy#7] [Ideas]
- [x] compressed list data structure 
- [ ] rds2py wrapper for compressed list
- [ ] migrating genomicrangeslist to use compressed lists

**Comment by hpages** [comment]
Interesting! For the record the CompressedList representation in Bioconductor started as a hack to work around the inefficiency of the more natural/straighforward SimpleList representation. With SimpleList you just create your individual S4 objects, store them in an ordinary list, and process that thing with `lapply` or `for` loops. However this approach is extremely inefficient when the list contains thousands or millions of S4 objects. Are you seeing this kind of bottlenecks in Python too? Just curious.

**Reply by jkanche** [reply]
I haven't done any benchmarks, but my guess is its going to be inefficient as well in python, example, its probably efficient to apply a function on a single numpy vector than a list of numpy vectors.

**Comment by jkanche** [comment]
also https://github.com/BiocPy/GenomicRanges/issues/148

**Comment by jkanche** [comment]
working on https://github.com/biocpy/compressed-lists
